verification/verifier_helpers.go

```diff
@@ -0,0 +1,53 @@
+package verification
+
+import (
+	"crypto/x509"
+	"fmt"
+	"strings"
+)
+
+// isFailureResult checks whether a VerificationResult fails the entire signature verification workflow.
+// signature verification workflow is considered failed if there is a VerificationResult with "Enforced" as the action but the result was inconclusive (failed to verify) or unsuccessful
+func isFailureResult(result VerificationResult) bool {
+	return result.Action == Enforced && (result.FailedToVerify || !result.Success)
+}
+
+func verifyX509TrustedIdentities(certs []*x509.Certificate, trustPolicy *TrustPolicy) error {
+	if isPresent(wildcard, trustPolicy.TrustedIdentities) {
+		return nil
+	}
+
+	var trustedX509Identities []map[string]string
+	for _, identity := range trustPolicy.TrustedIdentities {
+		i := strings.Index(identity, ":")
+
+		identityPrefix := identity[:i]
+		identityValue := identity[i+1:]
```

Comment:
`strings.Cut()` might be used instead.

Reply:
This surely makes the code look pretty. Will address in my next PR

Reply:
Looks like strings.Cut() was added in 1.18. notation-go is still on 1.17
We can address it when we upgrade to 1.18

Reply:
Please create an issue for the future update. Meanwhile, the index `i` needs to be checked to see whether `i >= 0`.

Reply:
We are already verifying that trusted identities are in the right format [here](https://github.com/notaryproject/notation-go/blob/main/verification/policy.go#L97)
Upgrade issue : https://github.com/notaryproject/notation-go/issues/75